hazelcast/src/test/java/com/hazelcast/executor/SingleNodeTest.java

```diff
@@ -0,0 +1,346 @@
+import java.util.concurrent.Future;
+import java.util.concurrent.RejectedExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
+
+@RunWith(HazelcastParallelClassRunner.class)
+@Category(QuickTest.class)
+public class SingleNodeTest extends ExecutorServiceTestSupport {
+
+    private IExecutorService executor;
+
+    @Before
+    public void setUp() {
+        executor = createSingleNodeExecutorService("test", 1);
+    }
+
+    @Test
+    public void hazelcastInstanceAwareAndLocal() throws Exception {
+        HazelcastInstanceAwareRunnable task = new HazelcastInstanceAwareRunnable();
+        try {
+            executor.submit(task).get();
+        } catch (ExecutionException e) {
+            throw (Error)e.getCause();
+        }
+    }
+
+    @Test(expected = NullPointerException.class)
+    public void submitNullTask() throws Exception {
+        executor.submit((Callable<?>)null);
+    }
+
+    @Test
+    public void testBasicTask() throws Exception {
+        Callable<String> task = new BasicTestTask();
+        Future future = executor.submit(task);
+        assertEquals(future.get(), BasicTestTask.RESULT);
+    }
+
+    @Test(expected = RejectedExecutionException.class)
+    public void testEmptyMemberSelector() {
+        HazelcastInstanceAwareRunnable task = new HazelcastInstanceAwareRunnable();
+        executor.execute(task, new MemberSelector() {
+            @Override
+            public boolean select(final Member member) {
+                return false;
+            }
+        });
+    }
+
+    @Test
+    public void testSubmitFailingCallableException_withExecutionCallback()
+            throws ExecutionException, InterruptedException
+    {
+        final CountDownLatch latch = new CountDownLatch(1);
+        executor.submit(new FailingTestTask(), new ExecutionCallback<String>() {
+            @Override
+            public void onResponse(String response) {
+            }
+
+            @Override
+            public void onFailure(Throwable t) {
+                latch.countDown();
+            }
+        });
+        assertTrue(latch.await(10, TimeUnit.SECONDS));
+    }
+
+    @Test(expected = CancellationException.class)
+    public void testCancellationAwareTask() throws ExecutionException, InterruptedException {
+        SleepingTask task = new SleepingTask(1);
+        Future future = executor.submit(task);
+        //noinspection EmptyCatchBlock
+        try {
+            future.get(1, TimeUnit.MILLISECONDS);
+            fail("Should throw TimeoutException!");
+        } catch (TimeoutException expected) {
+        }
+        assertFalse(future.isDone());
+        assertTrue(future.cancel(true));
+        assertTrue(future.isCancelled());
```

Comment:
You are both expecting an exception and making assertions. When the exception is not thrown, you fail with fail() call. Therefore, assert statements are not reached. 

Even if you don't call fail(), when you do not throw CancellationException, test will fail because of the absence of the exception. If it is also expected to fail with one of the assertions when the exception is not thrown, I think that test method can be divided into multiple test methods.